manager/controlapi/ca_rotation.go

```diff
@@ -0,0 +1,345 @@
+//    - If the desired cert is the same as the current CA cert, or is a different cert with the same key and subject, then abort
+//      any outstanding rotations. The current signing key is replaced with the desired signing key (this could lets us switch
+//      between external->internal or internal->external without an actual CA rotation, which is not needed because any leaf
+//      cert issued with one CA cert can be validated using the second CA certificate).
+//    - If the desired cert is the same as the current to-be-rotated-to CA cert, or is a different cert with the same key and
+//      subject, then a new root rotation is not needed. The current to-be-rotated-to signing key is replaced with the desired
+//      signing key (this could lets us switch between external->internal or internal->external without an actual CA rotation,
+//      which is not needed because any leaf cert issued with one CA cert can be validated using the second CA certificate).
+//    - Otherwise, start a new root rotation using the desired signing cert and desired signing key as the root rotation
+//      signing cert and key.  If a root rotation is already in progress, just replace it and start over.
+func validateCAConfig(ctx context.Context, securityConfig *ca.SecurityConfig, cluster *api.Cluster) (*api.RootCA, error) {
+	newConfig := cluster.Spec.CAConfig
+
+	if len(newConfig.SigningCAKey) > 0 && len(newConfig.SigningCACert) == 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument, "if a signing CA key is provided, the signing CA cert must also be provided")
+	}
+
+	if err := validateHasRequiredExternalCAs(ctx, securityConfig, cluster); err != nil {
+		return nil, err
+	}
+
+	// if the desired CA cert and key are not set, then we are happy with the current root CA configuration, unless
+	// the ForceRotate version has changed
+	if len(newConfig.SigningCACert) == 0 {
+		if cluster.RootCA.LastForcedRotation != newConfig.ForceRotate {
+			newRootCA, err := ca.CreateRootCA(ca.DefaultRootCN)
+			if err != nil {
+				return nil, grpc.Errorf(codes.Internal, err.Error())
+			}
+			return newRootRotationObject(ctx, securityConfig, cluster, newRootCA, newConfig.ForceRotate)
+		}
+		return &cluster.RootCA, nil // no change, return as is
+	}
+
+	// A desired cert and maybe key were provided - we need to make sure the cert and key (if provided) match.
+	var signingCert []byte
+	if newConfig.SigningCAKey != nil {
+		signingCert = newConfig.SigningCACert
+	}
+	newRootCA, err := ca.NewRootCA(newConfig.SigningCACert, signingCert, newConfig.SigningCAKey, ca.DefaultNodeCertExpiration, nil)
+	if err != nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	if len(newRootCA.Pool.Subjects()) != 1 {
+		return nil, grpc.Errorf(codes.InvalidArgument, "the desired CA certificate cannot contain multiple certificates")
+	}
+
+	// because newRootCA was parsed without error, this should not fail either
```

Comment:
Is this saying that `NewRootCA` parses `SigningCACert`, so we expect parsing it again will succeed?